### ISSUE TITLE
E0059, currying and body braces

### ISSUE THREAD

**Issue by kkkrist** [quick-lint/quick-lint-js#636]
Just discovered quick-lint-js today and I'm amazed by the speed of it. What a difference it makes!

One thing I stumbled upon:

This code will trigger a [E0057: use of undeclared variable](https://quick-lint-js.com/errors/#E0057) warning for `a`:

```javascript
const fn1 = a => b => { return a + b }
```

This does not:

```javascript
const fn2 = a => b => a + b
```
Is this intended?

**Comment by strager** [comment]
> Is this intended?

Nope!

Thanks for the bug report. I think this bug is caused by a refactor (8087d21b4ca2a7987821e20a606ca97b664bf610). I'll work on a fix.

**Comment by vegerot** [comment]
Nit: while there are pros and cons to [pre-releasing extensions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions) (your points about not wanting automatic codesigning in particular), one pro is that in cases like this you can just tell people to grab the beta version(assuming they’re using VSCode).

qljs is not hard to install manually which makes this less of an issue, but for the VSCode n00bs it can be nice 😊
